Initial layout for acceptance testing

DavidS opened the pull request:
The tests in this commit are already running successfully to the point
where the notify is causing a failure on the target agent.

The nodesets still need to be improved to use vagrant images instead of
the vmpooler reference.

jpogran commented on .sync.yml:
```diff
@@ -0,0 +1,26 @@
+.travis.yml:
+  script: "\"bundle exec rake validate && bundle exec rake lint && bundle exec rake spec SPEC_OPTS='--color --format documentation'\""
+  extras:
+  - rvm: 1.9.3
```
@puppetlabs/windows  We shouldn't include 1.9.3 and puppet 3.3.0 because we are targeting puppet 4 correct?

ferventcoder replied:
Correct. Just the earliest version of 4 that is supported.

jpogran commented on .sync.yml:
```diff
@@ -0,0 +1,26 @@
+appveyor.yml:
+  matrix_extras:
+    - PUPPET_GEM_VERSION: 3.3.0
+      RUBY_VER: 193
```
@puppetlabs/windows  We shouldn't include 1.9.3 and puppet 3.3.0 because we are targeting puppet 4 correct?

ferventcoder replied:
So this would be like

~~~yaml
 - PUPPET_GEM_VERSION: 4.0.0
   RUBY_VER: 200
~~~

glennsarti replied:
Default appveyor already has
```
    - PUPPET_GEM_VERSION: '~> 4.0'
      RUBY_VER: 21
    - PUPPET_GEM_VERSION: '~> 4.0'
      RUBY_VER: 21-x64
```

Does that cover ruby 2.0 and pup 4.0

ferventcoder commented:
👍  once you make those changes.

glennsarti commented on .sync.yml:
```diff
@@ -0,0 +1,26 @@
+NOTICE:
+  copyright_holders:
+    - name: 'Puppet, Inc.'
+      begin: 2012
```
Is 2012 the correct date?

ferventcoder replied:
That would be a no. If we are taking this over from someone else, we would need to include those folks as well.

DavidS committed:
Initial layout for acceptance testing

The tests in this commit are already running successfully to the point
where the notify is causing a failure on the target agent.

The nodesets still need to be improved to use vagrant images instead of
the vmpooler reference.

DavidS commented:
This is depending on #4 and should only be merged afterwards.

tphoney commented:
woo for tests !!! +1

DavidS commented:
restarted appveyor as it hung on rubygems download.

DavidS commented:
appveyor needed a second restart after hanging a build right after "Starting Build", but it is green now.
